src/app/Library/Widget.php

```diff
@@ -257,6 +256,11 @@ public function dump()
         return $this;
     }
 
+    public function crud()
+    {
+        return app('crud');
+    }
```

Comment:
Ah damn it... the `Widget` class was independent from the CRUD... theoretically... That was done intentionally, because Widgets can be used outside CRUD views too...

So if we move viewNamespaces to the `CrudPanel` object... and add this method here... they're no longer independent.

Reply:
Ok then... how about as an alternative, we DON'T add all viewNamespaces to the CrudPanel object. But instead... we add them to each individual object that needs them?

```php
CrudField {
    public $viewNamespaces;
    public static function hasViewNamespace($path);
    public static function addViewNamespace($path);
    public static function getViewNamespaces();
}

CrudColumn {
    public $viewNamespaces;
    public static function hasViewNamespace($path);
    public static function addViewNamespace($path);
    public static function getViewNamespaces();
}

CrudFilter {
    public $viewNamespaces;
    public static function hasViewNamespace($path);
    public static function addViewNamespace($path);
    public static function getViewNamespaces();
}

Widget {
    public $viewNamespaces;
    public static function hasViewNamespace($path);
    public static function addViewNamespace($path);
    public static function getViewNamespaces();
}
```

Which I guess could be structured into a trait, smth like:

```php
CrudField {
    use HasViewNamespaces;
    public $viewNamespacePrefix = 'fields';
}

CrudColumn {
    use HasViewNamespaces;
    public $viewNamespacePrefix = 'columns';
}

CrudFilter {
    use HasViewNamespaces;
    public $viewNamespacePrefix = 'filters';
}

Widget {
    use HasViewNamespaces;
    public $viewNamespacePrefix = 'widgets';
}

// provides a way to extend Backpack's "crud.view_namespaces" config item
trait hasViewNamespaces
{
    public $viewNamespaces;
    public static function hasViewNamespace($path);
    public static function addViewNamespace($path);
    public static function getViewNamespaces();
}
```

Reply:
I like the structure, but I am unsure if this would work out for us. 
The reason I think this should be done in `CrudPanel` it's because it's a singleton, we have only one instance during the app lifecycle, while we init multiple instances of `CrudField` etc. 

Thinking about the problem we want to solve, how in `BetterExports` I would register the new button view namespace? 

`CrudButton::addViewNamespace('better-exports::buttons')` would add the view namespace only to this `CrudButton` instance.

I thought about binding into `CrudButton` or something like, but I don't think it would work as we need something to be available on multiple providers (package providers). 

We did it in config because it's "globally" available, I switched to CrudPanel for the same reason. Am I missing something here ? I really like this organization structure, the most "similar" would be the panel property .. but we have the `Widget` issue. 

Well, about that issue ... Widget is a singleton too ... so we can add the `HasViewNamespaces` into the `Widget`, and into the `CrudPanel`? 

For fields/columns/buttons etc we do `Crud::addViewNamespace()` for widgets we do `Widget::addViewNamespace()` ? This seems like a good solution no ?

Reply:
Ah right. Didn't think of that. Good job Pedro.

Sounds like a plan! Let's do it 💪